map-storage/src/Services/CustomEntityCollectionService.ts

```diff
@@ -0,0 +1,113 @@
+import {
+    CollisionGrid,
+    ENTITIES_FOLDER_PATH,
+    ENTITY_COLLECTION_FILE,
+    EntityCollectionRaw,
+    EntityRawPrefab,
+    mapCustomEntityDirectionToDirection,
+} from "@workadventure/map-editor";
+import { DeleteCustomEntityMessage, ModifyCustomEntityMessage, UploadEntityMessage } from "@workadventure/messages";
+import { fileSystem } from "../fileSystem";
+import { mapPathUsingDomainWithPrefix } from "./PathMapper";
+
+export class CustomEntityCollectionService {
+    private readonly hostname: string;
+
+    constructor(hostname: string) {
+        this.hostname = hostname;
+    }
+
+    private getEntityCollectionFileVirtualPath() {
+        return mapPathUsingDomainWithPrefix(`${ENTITIES_FOLDER_PATH}/${ENTITY_COLLECTION_FILE}`, this.hostname);
+    }
+
+    private getEntityToUploadVirtualPath(fileName: string) {
+        return mapPathUsingDomainWithPrefix(`${ENTITIES_FOLDER_PATH}/${fileName}`, this.hostname);
+    }
+
+    public async uploadEntity(uploadEntityMessage: UploadEntityMessage) {
+        const { imagePath, file } = uploadEntityMessage;
+        await fileSystem.writeByteAsFile(this.getEntityToUploadVirtualPath(imagePath), file);
```

Comment:
Let's rename `writeByteAsFile` to `writeByteArrayAsFile` (we are not writing a single Byte :) )

Reply:
My bad :)